fnet/dial.go

```diff
@@ -4,31 +4,26 @@
 package fnet
 
 import (
-	"net"
+	"net/http"
 	"time"
 
 	"github.com/onsi/ginkgo/v2"
 	"github.com/onsi/gomega"
 )
 
-// DialAndRead dials the network address, reads the data from the established connection, and asserts it against want.
-func DialAndRead(network, address, want string, maxRetry int, retryInterval time.Duration) {
-	var (
-		conn net.Conn
-		err  error
-	)
+// HTTPGetAndAssert sends an HTTP GET request to the specified URL, asserts the response status code against want, and closes the response body.
+func HTTPGetAndAssert(url string, want int, maxRetry int, retryInterval time.Duration) {
+	var err error
 	for i := 0; i < maxRetry; i++ {
-		conn, err = net.Dial(network, address)
+		// #nosec G107 // it does not matter if url is not a constant for testing.
+		resp, err := http.Get(url)
 		if err != nil {
 			time.Sleep(retryInterval)
 			continue
 		}
 
-		b := make([]byte, len([]byte(want))) //nolint:makezero // The content of b does not matter,
-		// but len(b) must be equal to len([]byte(want)) so that conn.Read can read the whole thing.
-		gomega.Expect(conn.Read(b)).Error().ShouldNot(gomega.HaveOccurred())
-		gomega.Expect(b).To(gomega.Equal([]byte(want)))
-		gomega.Expect(conn.Close()).To(gomega.Succeed())
+		gomega.Expect(resp.StatusCode).To(gomega.Equal(want))
+		gomega.Expect(resp.Body.Close()).To(gomega.Succeed())
```

Comment:
nit: Usually `resp.Body.Close()` is `defer`ed right after checking that the `err` from the request is not `nil`. This ensures that the response body is closed no matter what happens later.

Reply:
Updated in https://github.com/runfinch/common-tests/pull/42